tests/components/greeneye_monitor/test_init.py

```diff
@@ -38,23 +36,88 @@
     assert_voltage_sensor_registered,
 )
 
+from tests.common import MockConfigEntry
+
 
-async def test_setup_fails_if_no_sensors_defined(
+async def test_setup_succeeds_no_config(
     hass: HomeAssistant, monitors: AsyncMock
 ) -> None:
-    """Test that component setup fails if there are no sensors defined in the YAML."""
-    success = await setup_greeneye_monitor_component_with_config(
-        hass, SINGLE_MONITOR_CONFIG_NO_SENSORS
+    """Test that component setup succeeds if there is no config present in the YAML."""
+    assert await async_setup_component(hass, DOMAIN, {})
+
+
+async def test_setup_creates_config_entry(
+    hass: HomeAssistant,
+    monitors: AsyncMock,
+) -> None:
+    """Test that component setup copies the YAML configuration into a config entry."""
+    assert await setup_greeneye_monitor_component_with_config(
+        hass, SINGLE_MONITOR_CONFIG_VOLTAGE_SENSORS
     )
-    assert not success
 
+    normalized_schema = CONFIG_SCHEMA(SINGLE_MONITOR_CONFIG_VOLTAGE_SENSORS)
 
-@pytest.mark.xfail(reason="Currently failing. Will fix in subsequent PR.")
-async def test_setup_succeeds_no_config(
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert len(entries) == 1
+    entry = entries[0]
+    assert entry.data == {CONF_PORT: normalized_schema[DOMAIN][CONF_PORT]}
+    assert entry.options == {CONF_MONITORS: normalized_schema[DOMAIN][CONF_MONITORS]}
+
+
+async def test_setup_from_config_entry(
```

Comment:
Can you add a test that has two monitors configured to catch what I think is the previously mentioned bug as well?

Reply:
Oddly, there is already a multi-monitor test, but it doesn't detect this issue. Looks like because the actual adding is happening in separate task, `await`ing the component setup doesn't detect it. If I add a test that actually tries to check the sensors, that catches it.